lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -2650,15 +2651,26 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
 
     // @_staticExclusiveOnly types cannot be put into 'var's, only 'let'.
     if (auto SD = VD->getInterfaceType()->getStructOrBoundGenericStruct()) {
-      if (SD->getAttrs().hasAttribute<StaticExclusiveOnlyAttr>() &&
-          !VD->isLet()) {
+      if (SD->getAttrs().hasAttribute<StaticExclusiveOnlyAttr>()) {
+        auto isProtocolContext = DC->getSelfProtocolDecl() != nullptr;
+
+        if (isProtocolContext && !VD->supportsMutation()) {
+          return;
+        }
```

Comment:
This also includes e.g computed properties in protocol extensions, is that intentional?

Reply:
Yes, computed properties cannot return `@_staticExclusiveOnly` types.

Reply:
To be clear, I'm saying that this would not reject that

Reply:
Thanks! Should be resolved by just directly asking the decl context if it's a protocol instead of going through `getSelfProtocolDecl`. Added a test case.